packages/sdk/android/src/main/java/com/primerioreactnative/components/manager/redirect/PrimerRNHeadlessUniversalCheckoutComponentWithRedirectManager.kt

```diff
@@ -0,0 +1,250 @@
+
+  @ReactMethod
+  fun configure(paymentMethodType: String, promise: Promise) {
+
+    val viewModelStoreOwner = reactContext.currentActivity as? ViewModelStoreOwner ?: run {
+      PrimerRNViewModelStoreOwner()
+    }
+
+    banksComponent =
+      PrimerHeadlessUniversalCheckoutComponentWithRedirectManager(viewModelStoreOwner).provide<BanksComponent>(
+        paymentMethodType = paymentMethodType
+      )
+
+    val lifecycleOwner = reactContext.currentActivity as? LifecycleOwner ?: run {
+      promise.reject("LifecycleOwner", "Expected activity class that implement LifecycleOwner")
+      return
+    }
+
+    job = lifecycleOwner.lifecycleScope.launch {
+      banksComponent?.start()
+      coroutineScope {
+        launch {
+          configureBanksListener()
+        }
+
+        launch {
+          configureValidationListener()
+        }
+
+        launch {
+          configureErrorListener()
+        }
+        promise.resolve(null)
+      }
+
+    }
+
+  }
+
+  @ReactMethod
+  fun onBankSelected(bankId: String, promise: Promise) {
+    banksComponent?.updateCollectedData(BanksCollectableData.BankId(bankId))
+    promise.resolve(null)
+
+  }
+
+  @ReactMethod
+  fun onBankFilterChange(filter: String, promise: Promise) {
+    banksComponent?.updateCollectedData(BanksCollectableData.Filter(filter))
+    promise.resolve(null)
+  }
+
+  private suspend fun configureErrorListener() {
+    banksComponent?.componentError?.collectLatest { error ->
+
+      sendEvent(
+        PrimerHeadlessUniversalCheckoutRedirectManagerEvent.ON_ERROR.eventName,
+        JSONObject().apply {
+          put(
+            "errors", JSONArray(
+              Json.encodeToString(error)
+            )
+          )
+        }
+      )
+    }
+  }
+
+  private suspend fun configureBanksListener() {
+    banksComponent?.componentStep?.collectLatest { banksStep ->
+      when (banksStep) {
+        is BanksStep.Loading -> {
+          sendEvent(
+            PrimerHeadlessUniversalCheckoutRedirectManagerEvent.ON_RETRIEVING.eventName,
+            JSONObject().apply {
+              put(
+                "retrieving", "true"
+              )
+            }
+          )
+        }
+
+        is BanksStep.BanksRetrieved -> {
+          sendEvent(
+            PrimerHeadlessUniversalCheckoutRedirectManagerEvent.ON_RETRIEVED.eventName,
+            JSONObject().apply {
+              put(
+                "banks", JSONArray(
+                  Json.encodeToString(banksStep.banks.map { it.toPrimerIssuingBankRN() })
+                )
+              )
+            }
+          )
+        }
+      }
+    }
+  }
+
+  private suspend fun configureValidationListener() {
+    banksComponent?.componentValidationStatus?.collectLatest { validationStatus ->
+      when (validationStatus) {
+        is PrimerValidationStatus.Validating -> {
+          sendEvent(PrimerHeadlessUniversalCheckoutRedirectManagerEvent.ON_VALIDATING.eventName,
+            JSONObject().apply {
```

Comment:
[q]: is there a need to send `validating` here?
Native SDKs send the data that is being validated, so we have a mismatch in DX:

``` 
data class Validating<T : PrimerCollectableData>(val collectableData: PrimerCollectableData) :
        PrimerValidationStatus<T>